addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,386 @@
+        }
+
+        @OnWebSocketConnect
+        public void onConnect(Session session) {
+            try {
+                logger.debug("Connect: {}", session.getRemoteAddress().getAddress());
+                ByteBuffer buf = generateRequest();
+                session.getRemote().sendBytes(buf);
+            } catch (IOException e) {
+                connectionError(e);
+            } catch (Exception e) {
+                logger.error("Error: {}", e.getMessage());
+            }
+        }
+
+        /**
+         * Method to generate ByteBuffer request to be sent to vallox online websocket
+         * to request or set data
+         *
+         * @param mode 246 for data request, 249 for setting data
+         * @param hmParameters HashMap for setting data with register as key and value as value
+         * @return ByteBuffer to be sent to websocket
+         */
+        public ByteBuffer generateCustomRequest(Integer mode, HashMap<Integer, Integer> hmParameters) {
+            // If we just want data the format is different so its just hardcoded here
+            if (mode == 246) {
+                // requestData (Length 3, Command to get data 246, empty set, checksum [sum of everything before])
+                return ByteBuffer.wrap(new byte[] { 3, 0, (byte) 246, 0, 0, 0, (byte) 249, 0 });
+            }
+
+            int numberParameters = (hmParameters.size() + 1) * 2; // Parameters + Checksum
+            int checksum = numberParameters;
+            // Allocate for Content incl. Checksum + Length
+            ByteBuffer bb = ByteBuffer.allocate((numberParameters + 1) * 2)
+                    .put(convertIntegerIntoByteBuffer(numberParameters));
+
+            // Put Mode, HashMap and checksum to ByteArray
+            bb.put(convertIntegerIntoByteBuffer(mode));
+            checksum += mode;
+
+            for (Map.Entry<Integer, Integer> i : hmParameters.entrySet()) {
+                bb.put(convertIntegerIntoByteBuffer(i.getKey()));
+                bb.put(convertIntegerIntoByteBuffer(i.getValue()));
+                checksum += i.getKey() + i.getValue();
+            }
+
+            bb.put(convertIntegerIntoByteBuffer(checksum));
+            bb.position(0);
+            return bb;
+        }
+
+        // Convert Integer to ByteBuffer in Little-endian
+        public ByteBuffer convertIntegerIntoByteBuffer(Integer i) {
+            byte b1 = (byte) (i % 256);
+            byte b2 = (byte) ((i - i % 256) / 256);
+
+            return ByteBuffer.wrap(new byte[] { b1, b2 });
+        }
+
+        public ByteBuffer generateRequest() {
+            if (updateState == null || channelUID == null) {
+                // requestData (Length 3, Command to get data 246, empty set, checksum [sum of everything before])
+                return generateCustomRequest(246, new HashMap<Integer, Integer>());
+            }
+            if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_STATE)) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)